File: Source/Plugins/Server/com.tle.core.notification/src/com/tle/core/notification/NotificationExtension.java
package com.tle.core.notification;

import com.google.common.collect.ListMultimap;
import com.tle.core.notification.beans.Notification;

public interface NotificationExtension
{
	boolean isIndexed(String type);

	boolean isForceEmail(String type);

	String emailText(ListMultimap<String, Notification> typeMap);

}

File: Source/Plugins/Notifications/com.tle.core.notification/scalasrc/com/tle/core/notification/NotificationExtension.scala
package com.tle.core.notification

import com.tle.common.usermanagement.user.valuebean.UserBean
import com.tle.core.notification.beans.Notification

trait NotificationExtension {

  def isIndexed(ntype: String): Boolean

  def emails(user: UserBean, notifications: Iterable[Notification], totals: Map[String, Int]): Iterable[NotificationEmail]
}
